src/rows-data.cpp

```diff
@@ -0,0 +1,121 @@
+#include <Rcpp.h>
+#include <dplyr.h>
+#include "utils.h"
+#include "rows-data.h"
+
+namespace Slices {
+
+
+CollationType hash_collate(std::string collate) {
+  if (collate == "rows")
+    return rows;
+  else if (collate == "cols")
+    return cols;
+  else
+    return list;
+}
+
+Settings::Settings(Environment execution_env_)
+    : output_colname(as<std::string>(execution_env_[".to"])),
+      include_labels(execution_env_[".labels"]) {
+  collation = hash_collate(as<std::string>(execution_env_[".collate"]));
+}
+
+
+Labels::Labels(Environment execution_env_)
+    : are_unique(execution_env_[".unique_labels"]),
+      slicing_cols(execution_env_[".slicing_cols"]),
+      labels_(execution_env_[".labels_cols"]),
+      n_labels_(Rf_length(execution_env_[".labels_cols"])) {
+}
+
+void Labels::remove(std::vector<int>& index) {
```

Comment:
Could this be `const std::vector<int>& index`? I think it's slightly preferred stylistically to use const-ref for objects that are read-only, and pass by pointer if the object will be modified.

This makes it very clear at the call site that an object could be mutated, eg:

```cpp
std::vector<int> x;
fill_vector(&x);
```

In the above call, one can infer that the `x` object is modified by `fill_vector()`; it is somewhat less obvious if that function were to accept just a plain reference.